### PR TITLE
Change quick info request to async for ST3

### PR DESCRIPTION
**Pull request by zhengbli**
Fix issue #216

### CODE REVIEW

File: typescript/libs/service_proxy.py [hunk 1]
```diff
@@ -155,11 +156,18 @@ def quick_info(self, path, location=Location(1, 1), onCompleted=None):
...
-            req_dict["seq"]
-        )
+        if not IS_ST2:
+            self.__comm.sendCmdAsync(
+                json_str,
+                req_dict["seq"],
+                lambda json_dict: None if onCompleted is None else onCompleted(json_dict)
+            )
+        else:
+            self.__comm.sendCmd(
```
Comment (DanielRosenwasser) [inline typescript/libs/service_proxy.py]: Is there any reason why `sendCmdAsync` and `sendCmd` don't take arguments in the same order?

File: typescript/libs/service_proxy.py [hunk 2]
```diff
@@ -155,11 +156,18 @@ def quick_info(self, path, location=Location(1, 1), onCompleted=None):
...
-        self.__comm.sendCmd(
-            lambda json_dict: None if onCompleted is None else onCompleted(json_dict),
-            json_str,
-            req_dict["seq"]
-        )
+        if not IS_ST2:
+            self.__comm.sendCmdAsync(
+                json_str,
+                req_dict["seq"],
+                lambda json_dict: None if onCompleted is None else onCompleted(json_dict)
```
Comment (DanielRosenwasser) [inline typescript/libs/service_proxy.py]: Factor the callback out so that you're not duplicating code.

Comment (DanielRosenwasser) [inline typescript/libs/service_proxy.py]: Also consider making this `onCompleted and onCompleted(json_dict)`.

Though I'm not sure how idiomatic that is in Python.

File: typescript/libs/service_proxy.py (hunk 1 above)
Comment (zhengbli) [inline typescript/libs/service_proxy.py]: It just happened to be that way, unifying it.

File: typescript/libs/service_proxy.py [hunk 3]
```diff
@@ -155,11 +156,19 @@ def quick_info(self, path, location=Location(1, 1), onCompleted=None):
         args = {"file": path, "line": location.line, "offset": location.offset}
         req_dict = self.create_req_dict("quickinfo", args)
         json_str = json_helpers.encode(req_dict)
-        self.__comm.sendCmd(
-            lambda json_dict: None if onCompleted is None else onCompleted(json_dict),
-            json_str,
-            req_dict["seq"]
-        )
+        callback = lambda json_dict: None if onCompleted is None else onCompleted(json_dict)
```
Comment (DanielRosenwasser) [inline typescript/libs/service_proxy.py]: What about just passing in `onCompleted or lambda: None` here as well as in all other locations?

File: typescript/libs/service_proxy.py [hunk 4]
```diff
@@ -45,9 +46,9 @@ def completions(self, path, location=Location(1, 1), prefix="", on_completed=Non
         req_dict = self.create_req_dict("completions", args)
         json_str = json_helpers.encode(req_dict)
         self.__comm.sendCmd(
-            lambda json_dict: None if on_completed is None else on_completed(json_dict),
             json_str,
-            req_dict["seq"]
+            req_dict["seq"],
+            lambda json_dict: None if on_completed is None else on_completed(json_dict)
```
Comment (DanielRosenwasser) [inline typescript/libs/service_proxy.py]: Naming here isn't consistent.

File: typescript/libs/service_proxy.py (hunk 3 above)
Comment (zhengbli) [inline typescript/libs/service_proxy.py]: I didn't quite understand last time you mentioned it. Now I got it :). That would be more concise.

Comment (DanielRosenwasser) [inline typescript/libs/service_proxy.py]: That's okay, the last thing I suggested was a lot stupider :smile:

**Comment by DanielRosenwasser** [comment]
:+1:

File: typescript/libs/node_client.py [hunk 5]
```diff
@@ -111,7 +111,7 @@ def makeTimeoutMsg(self, cmd, seq):
         }
         return timeoutMsg
 
-    def sendCmd(self, cb, cmd, seq):
+    def sendCmd(self, cmd, seq, cb):
```
Comment (billti) [inline typescript/libs/node_client.py]: You have derived a class from CommClient and overridden this sendCmd method, you have have changed the meaning of the first 3 params (which are "self, cmd, cb" in the base class).  You should keep the matching parameters the same in the derived class (as they were in the original code).

Comment (zhengbli) [inline typescript/libs/node_client.py]: Changed that to match the same order. The original code seemed wrong too, as `seq` appears in the middle. I think it won't affect the functionality as the number of parameters is different, but I agree for consistence they should be kept the same order.

**Pull request merged**